benches/state_space.rs

```diff
@@ -1,5 +1,104 @@
-use criterion as _;
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
 
-fn main() {
-    todo!("Implement benchmarks for `state_space` module!");
+fn criterion_benchmark(c: &mut Criterion) {
+    c.bench_function("add state changes benchmark", |b| {
+        b.iter(|| bench_funcs::test_add_state_changes(black_box(3)));
+    });
+    c.bench_function("add empty state changes benchmark", |b| {
+        b.iter(|| bench_funcs::test_add_empty_state_changes(black_box(3)));
+    });
+}
+
+criterion_group!(benches, criterion_benchmark);
+criterion_main!(benches);
+
+mod bench_funcs {
```

Comment:
Lets remove this additional module and just put any helper functions directly the existing mod.